worlds/__init__.py

```diff
@@ -122,6 +119,26 @@ def load(self) -> bool:
             elif entry.is_file() and entry.name.endswith(".apworld"):
                 world_sources.append(WorldSource(file_name, is_zip=True, relative=relative))
 
+# Function to load worlds from a zip archive
+def load_worlds_from_zip(zip_file: str):
+    zip_file = zip_file.split(".zip")[0] + ".zip"
+    with zipfile.ZipFile(zip_file, 'r') as z:
+        for file_name in z.namelist():
+            if not file_name.startswith("worlds/"):
+                continue
+            file_name = file_name.split("worlds/")[1]
+            if not file_name.startswith(("_", ".")):
+                world_sources.append(WorldSource(file_name, is_zip=True, relative=True))
+
+# find potential world containers, currently folders and zip-importable .apworld's
+world_sources: List[WorldSource] = []
+for folder in (folder for folder in (user_folder, local_folder) if folder):
+    if ".zip" in folder:
```

Comment:
this will only catch when running via traditionally named zip files. I couldn't find a way to detect from the python runtime if we're running from within a zip. Open to suggestions here.